backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java

```diff
@@ -0,0 +1,355 @@
+		invoiceCandidate.setC_Flatrate_Term_ID(modularContract.getC_Flatrate_Term_ID());
+		invoiceCandidate.setC_Async_Batch_ID(modularContract.getC_Async_Batch_ID());
+
+		InvoiceCandidateLocationAdapterFactory
+				.billLocationAdapter(invoiceCandidate)
+				.setFrom(ContractLocationHelper.extractBillLocation(modularContract));
+
+		invoiceCandidate.setIsTaxIncluded(modularContract.isTaxIncluded());
+
+		//
+		invoiceCandidate.setLine(moduleConfig.getSeqNo().toInt());
+		invoiceCandidate.setM_Product_ID(moduleConfig.getProductId().getRepoId());
+		invoiceCandidate.setInvoicingGroup(moduleConfig.getInvoicingGroup());
+		invoiceCandidate.setProductName(moduleConfig.getName());
+		//
+		invoiceCandidate.setHarvesting_Year_ID(yearAndCalendarId.yearId().getRepoId());
+		invoiceCandidate.setC_Harvesting_Calendar_ID(yearAndCalendarId.calendarId().getRepoId());
+		//
+		invoiceCandidate.setM_PricingSystem_ID(PricingSystemId.toRepoId(createInvoiceCandidateRequest.getPricingSystemId()));
+		//
+		invoiceCandidate.setModCntr_Module_ID(moduleConfig.getId().getModularContractModuleId().getRepoId());
+
+		final ComputingResponse computingResponse = getComputingMethodResponse(createInvoiceCandidateRequest);
+
+		setPriceAndQty(invoiceCandidate, computingResponse);
+
+		trxManager.runAfterCommit(() -> modularContractLogDAO.setICProcessed(ModularContractLogQuery.ofEntryIds(computingResponse.getIds()),
+				InvoiceCandidateId.ofRepoId(invoiceCandidate.getC_Invoice_Candidate_ID())));
+
+		return invoiceCandidate;
+	}
+
+	private void setPriceAndQty(
+			@NonNull final I_C_Invoice_Candidate invoiceCandidate,
+			@NonNull final ComputingResponse computingResponse)
+	{
+		final ProductPrice productPrice = computingResponse.getPrice();
+		invoiceCandidate.setC_Currency_ID(CurrencyId.toRepoId(productPrice.getCurrencyId()));
+		invoiceCandidate.setPriceActual(productPrice.toBigDecimal());
+		invoiceCandidate.setPriceEntered(productPrice.toBigDecimal());
+		invoiceCandidate.setPrice_UOM_ID(UomId.toRepoId(productPrice.getUomId()));
+
+		invoiceCandidate.setQtyEntered(computingResponse.getQty().toBigDecimal());
+		invoiceCandidate.setC_UOM_ID(computingResponse.getQty().getUomId().getRepoId());
+	}
+
+	@NonNull
+	private ComputingResponse getComputingMethodResponse(@NonNull final CreateInvoiceCandidateRequest createInvoiceCandidateRequest)
+	{
+		final I_C_Flatrate_Term modularContract = createInvoiceCandidateRequest.getModularContract();
+		final FlatrateTermId flatrateTermId = FlatrateTermId.ofRepoId(modularContract.getC_Flatrate_Term_ID());
+		final ModuleConfig moduleConfig = createInvoiceCandidateRequest.getModuleConfig();
+
+		final ComputingMethodType computingMethodType = moduleConfig.getComputingMethodType()
+				.orElseThrow(() -> new AdempiereException("No Computing Method Type found for ModuleConfig !")
+						.appendParametersToMessage()
+						.setParameter("ModuleConfigId", moduleConfig.getId().getRepoId()));
```

Comment:
Shouldn't be needed as it's mandatory now, pls adjust in `de.metas.contracts.modular.settings.ModularContractSettingsDAO#fromPOs` ,`de.metas.contracts.modular.settings.ModularContractType` and other places you can find